src/web/kestraFsProvider.ts

```diff
@@ -0,0 +1,206 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+
+import * as path from 'path';
+import * as vscode from 'vscode';
+
+type KestraFileAttributes = {
+	fileName: string;
+	lastModifiedTime: number;
+	creationTime: number;
+	type: keyof typeof vscode.FileType;
+	size: number;
+};
+
+const fileStatFromKestraFileAttrs = ({ fileName, type, creationTime, lastModifiedTime, size }: KestraFileAttributes): vscode.FileStat => {
+	const finalStats = {
+		type: vscode.FileType[type],
+		ctime: creationTime,
+		mtime: lastModifiedTime,
+		size
+	};
+
+	return finalStats;
+};
+
+const EXCLUDED_FOLDERS = [".git", ".vscode"];
+
+export class KestraFS implements vscode.FileSystemProvider {
+	namespace: string;
+	url: string;
+
+	constructor(namespace: string) {
+		this.namespace = namespace;
+		this.url = vscode.workspace.getConfiguration("kestra.api").get("url") as string;
+	}
+
+	private async callFileApi(suffix?: string, options?: RequestInit): Promise<Response> {
+		const fetchResponse = await fetch(`${this.url}/api/v1/files/${this.namespace}${suffix ?? ""}`, options);
+		if(fetchResponse.status === 404) {
+			throw vscode.FileSystemError.FileNotFound(suffix);
+		}
+		return fetchResponse;
+	}
+
+	private async callFlowsApi(suffix?: string, options?: RequestInit): Promise<Response> {
+		const fetchResponse = await fetch(`${this.url}/api/v1/flows${suffix ?? ""}`, options);
+		if(fetchResponse.status === 404) {
+			throw vscode.FileSystemError.FileNotFound(suffix);
+		}
+		return fetchResponse;
+	}
+
+	private isExcludedFolder(uri: vscode.Uri) {
+		return EXCLUDED_FOLDERS.some(f => uri.path.includes(f));
+	}
+
+	private trimNamespace(path: string) {
+		return path.substring(this.namespace.length + 1);
+	}
+
+	private isFlow(uri: vscode.Uri) {
+		return uri.path.startsWith(`/${this.namespace}/flows/`);
+	}
+
+	private isFlowsDirectory(uri: vscode.Uri) {
+		return uri.path === `/${this.namespace}/flows`;
+	}
+
+	private impactsFlowsDirectory(uri?: vscode.Uri) {
+		if(!uri) {
+			return false;
+		}
+
+		return this.isFlowsDirectory(uri) || this.isFlow(uri);
+	}
+
+	private extractFlowId(uri: vscode.Uri): string {
+		return uri.path.substring(uri.path.lastIndexOf("/") + 1);
+	}
+
+	private async getFlowSource(uri: vscode.Uri): Promise<string> {
+		const flowId = this.extractFlowId(uri);
+		return (await (await this.callFlowsApi(`/${this.namespace}/${flowId}?source=true`)).json()).source;
+	}
+
+	async stat(uri: vscode.Uri): Promise<vscode.FileStat> {
```

Comment:
Those are more metadata instead of Stats ?

Reply:
That's the FileProviderSystem interface that brings this :P

Reply:
Wasn't sure, but you confirm it thanks